File: java-codegen/src/main/java/org/opensearch/client/codegen/OpenSearchJavaCodegenPlugin.java
package org.opensearch.client.codegen;

import software.amazon.smithy.build.PluginContext;
import software.amazon.smithy.build.SmithyBuildPlugin;

public class OpenSearchJavaCodegenPlugin implements SmithyBuildPlugin {
    @Override
    public String getName() {
        return "opensearch-java";
    }

    @Override
    public void execute(PluginContext context) {
        new CodegenVisitor(context).execute();
    }
}

File: java-codegen/src/main/kotlin/org/opensearch/client/codegen/OpenSearchJavaCodegenPlugin.kt
package org.opensearch.client.codegen

import software.amazon.smithy.build.PluginContext
import software.amazon.smithy.build.SmithyBuildPlugin

class OpenSearchJavaCodegenPlugin : SmithyBuildPlugin {
    override fun getName(): String = "opensearch-java"

    override fun execute(context: PluginContext) {
        CodegenVisitor(context).execute()
    }
}